### PR TITLE
Ensure saved model input signature contains only required features

### PR DESCRIPTION
**Pull request by oliverholworthy**
Relates to #898 

### Goals :soccer:

Ensure that the input signature of a saved merlin model contains only those required for the inputs of the model

### Implementation Details :construction:

- The saved model signautre generated by keras/tensorflow includes all features passed to the model at training time, even if some are ignored by the input layers.
- Adding a property/setter for schema on DataLoader to enable changing of the loader schema after it's initialized
- If a schema is provided to the `InputBlockV2`. Use this to set the schema on the loader.
  - This ensures that the loaded will only return required features and pass them to the fit method at training time.

### Testing Details :mag:

- Adds a test that checks the input signature with an inputblock that has a schema with fewer input features than the dataset

### CODE REVIEW

**Comment by jperez999** [comment]
rerun tests

**Comment by karlhigley** [comment]
I'm not the right reviewer for this since I'm not very familiar with the Models code base, but assuming it does what it says on the tin (and I trust that @oliverholworthy's work does so more than just about anyone else's including mine), it sounds like a reasonable solution to me.

File: merlin/models/tf/outputs/topk.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def extract_ids_embeddings(self, data: merlin.io.Dataset, check_unique_ids: bool
         if check_unique_ids:
             self._check_unique_ids(data=data)
         values = tf_utils.df_to_tensor(data)
-        ids = tf_utils.df_to_tensor(data.index)
+        ids = tf_utils.df_to_tensor(make_df({"index": data.index}))
```
Comment (oliverholworthy) [inline merlin/models/tf/outputs/topk.py:103]: This breaks with the newer versions of cudf. A RangeIndex no longer has an `empty` property and won't work with this df_to_tensor function.

**Comment by oliverholworthy** [comment]
This PR is ready to review/merge now.

**Review by edknv — APPROVED** [review]

**Pull request merged**